src/Core/tests/Unit/Stream/FixedSizeStreamTest.php

```diff
@@ -43,6 +43,16 @@ public function testChunk(RequestStream $content, int $size, array $expected): v
         self::assertSame($expected, \iterator_to_array($stream));
     }
 
+    /**
+     * @dataProvider provideChunks
+     */
+    public function testDecoratingFixedSize(RequestStream $content, int $size, array $expected): void
+    {
+        $stream = FixedSizeStream::create(FixedSizeStream::create($content, \random_int(1, 10)), $size);
```

Comment:
We don’t need the inner stream to be random here. Do we?

We don’t want our tests to fail randomly

Reply:
it don't have to be random, but it should work whatever the value of the inner chunk size.
if the test failed randomly, then their is a bug

Reply:
well, we should have 2 non-random tests: one using the same size, and one using always a different size.

Reply:
removed.